src/main/java/org/folio/innreach/domain/service/impl/ReferenceDataLoader.java

```diff
@@ -0,0 +1,71 @@
+package org.folio.innreach.domain.service.impl;
+
+import static org.apache.commons.collections4.CollectionUtils.isEmpty;
+
+import java.io.IOException;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+import org.springframework.stereotype.Service;
+
+import org.folio.innreach.client.InstanceContributorTypeClient;
+import org.folio.innreach.client.InstanceTypeClient;
+import org.folio.innreach.util.JsonHelper;
+
+@Log4j2
+@Service
+@RequiredArgsConstructor
+public class ReferenceDataLoader {
+
+  private static final String BASE_DIR = "reference-data";
+  private static final String INSTANCE_TYPES_DIR = "instance-types";
+  private static final String CONTRIBUTION_NAME_TYPES_DIR = "contributor-name-types";
+
+  private static final PathMatchingResourcePatternResolver resourcePatternResolver =
+    new PathMatchingResourcePatternResolver(ReferenceDataLoader.class.getClassLoader());
+
+  private final InstanceContributorTypeClient instanceContributorTypeClient;
+  private final InstanceTypeClient instanceTypeClient;
+  private final JsonHelper jsonHelper;
+
+  public void loadRefData() {
+    try {
+      log.info("Loading reference data");
+      loadInstanceTypes();
```

Comment:
not sure this is a good idea to populate data in other modules during tenant initialization

Reply:
Updated service to run asynchronously with retries. This should make it much more reliable